streams-bootstrap-cli/src/main/java/com/bakdata/kafka/KafkaApplication.java

```diff
@@ -299,20 +294,16 @@ public final CleanableApp<CR> createCleanableApp() {
     protected abstract CA createConfiguredApp(final A app, AppConfiguration<T> configuration);
 
     /**
-     * Configure application when running in debug mode. By default, Log4j2 log level is configured to debug for
-     * {@code com.bakdata} and the applications package.
+     * Called before starting the application, e.g., invoking {@link #run()}
      */
-    protected void configureDebug() {
-        Configurator.setLevel("com.bakdata", Level.DEBUG);
-        Configurator.setLevel(this.getClass().getPackageName(), Level.DEBUG);
+    protected void onApplicationStart() {
```

Comment:
I'm still not entirely sure when we opt for methods that we can overwrite (like `createTopicConfig`) and when for interfaces that can be passed (like the clean up hooks)

Reply:
This is the base class for all cli apps and it is an abstract class so I opted for a method that can be overriden. Adding some kind of interface here makes things much more confusing I think. Have a field that is set via constructor of the abstract class? Return an instance of the interface instead of just calling the method? What other options are there?